### PR TITLE
User story 11

### PR DESCRIPTION
**Pull request by leiyakenney**
- Adds merchant employee capabilities
- Writes tests for merchant navigation restrictions

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,7 +1,11 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  helper_method :cart, :current_user
+  helper_method :cart
+  helper_method :current_user
```
Comment (nancylee713) [inline app/controllers/application_controller.rb]: Can we also concatenate these methods into a single line, like
`helper_method :cart, :current_user, ...` ?

File: app/controllers/merchant/base_controller.rb [hunk 2]
```diff
@@ -1,7 +1,12 @@
 class Merchant::BaseController < ApplicationController
+  before_action :require_merchant_employee
```
Comment (nancylee713) [inline app/controllers/merchant/base_controller.rb]: Same thing here, we could do `before_action :method_1, :method_2`

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@ def new
       flash[:notice] = "You are already logged in"
       if user.admin_user?
         redirect_to "/admin"
-      elsif user.merchant_admin?
+      elsif user.merchant_admin? || user.merchant_employee?
```
Comment (nancylee713) [inline app/controllers/sessions_controller.rb]: Just to double check, did we change the role name to `merchant_admin` from `merchant_user` ?

File: spec/features/navigation_spec.rb [hunk 4]
```diff
@@ -112,4 +101,88 @@
...
+        address: "175 Beet Rd",
+        city: "Scranton",
+        state: "PA",
+        zipcode: "18501",
+        email: "dwightkschrute@email.com",
+        password: "IdentityTheftIsNotAJoke",
+        role: 1)
+    end
+
+    it "I see profile, logout, and merchant dashboard links on navigation bar" do
```
Comment (nancylee713) [inline spec/features/navigation_spec.rb]: Merchant dashboard looks good!

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -8,7 +8,7 @@ def new
       flash[:notice] = "You are already logged in"
       if user.admin_user?
         redirect_to "/admin"
```
Comment (nancylee713) [inline app/controllers/sessions_controller.rb]: A minor thing => can we change this to `admin_path` just to practice using a path helper?

**Comment by MackHalliday** [comment]
Looks good. 👯

**Pull request merged**